kamon-akka-http/src/main/scala/kamon/akka/http/instrumentation/ClientRequestInstrumentation.scala

```diff
@@ -55,9 +55,14 @@ class ClientRequestInstrumentation {
         if(status.isFailure())
           span.addError(status.reason())
 
-        span
-          .tag("http.status_code", response.status.intValue())
-          .finish()
+        val spanWithStatusTag = span.tag("http.status_code", status.intValue())
+        val spanWithStatusMetricTag = if (AkkaHttp.addHttpStatusCodeAsMetricTag) {
```

Comment:
When adding a metric tag to a Span that metric is also added as a Span tag so you could simplify this into a simple if/else on the setting value.

Reply:
Ahh, good point!